Remove the kDNSServiceFlagsTimeout flag when calling DNSServiceGetAdd…

nivi-apple opened the pull request:
…rInfo

- We should not be terminating the queries if we do not get anything back quickly The queries should be running until we get something back

Fixes: #33051

nivi-apple committed:
Remove the kDNSServiceFlagsTimeout flag when calling DNSServiceGetAddrInfo

- We should not be terminating the queries if we do not get anything back quickly
  The queries should be running until we get something back

nivi-apple commented on src/platform/Darwin/DnssdImpl.cpp:
```diff
@@ -43,7 +43,7 @@ constexpr uint16_t kSRPTimeoutInMsec = 250;
 
 constexpr DNSServiceFlags kRegisterFlags        = kDNSServiceFlagsNoAutoRename;
 constexpr DNSServiceFlags kBrowseFlags          = kDNSServiceFlagsShareConnection;
```
Do we need these constants kBrowseFlags, kGetAddrInfoFlags? could just use the kDNSServiceFlagsShareConnection in the api calls? we might want to keep them for future use cases where we might add to the flags

ksperling-apple replied:
I agree, I feel like seeing the flags inline at the call site would be more helpful than having these constants.

kiel-apple replied:
It looks like `kRegisterFlags` is the only one of these constants that ends up being used more than once, and it's reused within a single function.

A middle ground would be to scope these constants to the functions where they are used rather than putting them in global scope, and that seems like the right solution for `kRegisterFlags` - the others are probably more clear with the flags inline.

nivi-apple replied:
I would also add the flag for register and reconfirm records as well in the call site itself to be consistent since it doesn't have multiple flags there. If folks disagree I will revert that

nivi-apple committed:
Remove the constants for the DNSServiceFlags

- Either use the DNSServiceFlags value directly in the DNS SD API calls or define a
  local scoped variable

kiel-apple commented on src/platform/Darwin/DnssdImpl.cpp:
```diff
@@ -179,10 +173,11 @@ CHIP_ERROR Register(void * context, DnssdPublishCallback callback, uint32_t inte
     ChipLogProgress(Discovery, "Registering service %s on host %s with port %u and type: %s on interface id: %" PRIu32,
                     StringOrNullMarker(name), StringOrNullMarker(hostname), port, StringOrNullMarker(type), interfaceId);
 
+    DNSServiceFlags registerFlags = kDNSServiceFlagsNoAutoRename;
```
this can be `const` here to avoid changes within scope

nivi-apple committed:
Make the registerFlags const

restyled-commits committed:
Restyled by clang-format

bzbarsky-apple commented on src/platform/Darwin/DnssdImpl.cpp:
```diff
@@ -179,10 +173,11 @@ CHIP_ERROR Register(void * context, DnssdPublishCallback callback, uint32_t inte
                     StringOrNullMarker(name), StringOrNullMarker(hostname), port, StringOrNullMarker(type), interfaceId);
 
-    RegisterContext * sdCtx = nullptr;
+    const DNSServiceFlags registerFlags = kDNSServiceFlagsNoAutoRename;
```
Why not constexpr?

nivi-apple replied:
done.

nivi-apple committed:
Address review comment

restyled-commits committed:
Restyled by clang-format
